### DISCUSSION TITLE
Build for iOS failed and suggested pod repo update. How to tell CodeMagic do a pod repo update?

### DISCUSSION THREAD

**Discussion by wyxcoder** [codemagic-ci-cd/codemagic-docs#2363] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Error: CocoaPods's specs repository is too out-of-date to satisfy dependencies.
To update the CocoaPods specs, run:
  pod repo update

### Expected results

Why not CodeMagic just do a pod repo update?

### Actual results

Failed

**Comment by icarusdust** [comment]
Hi, regarding your question about why Codemagic does not do it, Codemagic won't run `pod repo update ` unless you ask for it just like other commands. Same as your local builds, if your local builds fail, your local machine does not update it automatically, only after you run `pod update`. 
As for the issue, you can add `pod repo update` as a shell script which should be executed before running the build command. Alternatively, you can check in the logs which library needs to be updated, and then open your `Podfile` or `Podfile.lock` to make the required updates.

**Comment by icarusdust** [comment]
I am closing the thread due to inactivity. Let me know if you have any questions by replying to this thread or opening a new one.
